### ISSUE TITLE
Changes.deleted returns empty list unless the assigned widget is visible in a ListView

### ISSUE THREAD

**Issue by n-doton** [realm/realm-dart#1208]
When listening to changes.inserted and changes.deleted from RealmResultsChanges, the changes.deleted list is only populated if the widget displaying the document data is currently built and visible within a ListView. When listening to changes.inserted, the list is populated regardless of if the widget is built or not.

Is this expected behaviour, and if so, is it possible to listen to all deletion indices?

**Comment by nielsenko** [comment]
No, there are no differences between inserted, modified, or deleted indexes. 

If it is consumed by a list view or something else doesn't matter.

But be sure to hold on to the `StreamSubscription` (`ListView` will do that for you) if you do a raw listen.

**Comment by n-doton** [comment]
Thanks for the clarification. I've just come back to this and seem to be getting the deletion index now.
